tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -386,6 +383,31 @@ private bool WriteAndEndResponse(BlockingAction blockingAction)
         return dict;
     }
 
+    public Dictionary<string, string[]> GetResponseHeadersForWaf()
+    {
+        var response = _context.Response;
+        var headersDic = new Dictionary<string, string[]>(response.Headers.Keys.Count);
```

Comment:
[Non-blocking] You can create the dictionary with a comparer that ignores case instead of calling `ToLowerInvariant()` on every key (which can potentially allocate a new string for each key).
```
var headersDic = new Dictionary<string, string[]>(
    response.Headers.Keys.Count,
    StringComparer.InvariantCultureIgnoreCase);
```

Reply:
We have a rule that the cookie keys need to be normalized to lower case before being passed to the WAF, so we what to change the case rather than ignoring the differences.

Reply:
Got it!